### PR TITLE
test: add more package tree tests

### PR DESCRIPTION
**Pull request by lee-tammy**
1. Tested if the data field of PackageTree replaced with Points of interests and added an end-to-end test for generating the package tree.
2. Fixed one of the tests in test-analysis that was failing.

### CODE REVIEW

File: test/test-package-tree.ts [hunk 1]
```diff
@@ -13,35 +13,31 @@
...
 import * as tree from '../src/package-tree';
-import {generatePackageTree, getPackageTreeFromDependencyList, PackageTree} from '../src/package-tree';
 
 import * as tests from './mock-projects';
+import {TestProject} from './util';
 
 test(
-    'the data property of the package tree should changed from undefined ' +
-        'to an array of Points of Interest',
+    'he data property of the package tree should changed from undefined ' +
```
Comment (kjin) [inline test/test-package-tree.ts]: Nit: `the`

File: test/test-package-tree.ts [hunk 2]
```diff
@@ -70,23 +66,168 @@ test(
...
+      t.deepEqual(c1Pjson.name, 'c');
+      t.deepEqual(c1Pjson.version, '1.0.0');
+      const c1Dependencies = b1Dependencies[0].dependencies.sort(sortByName);
+      const d1Pjson =
+          require(path.join(c1Dependencies[0].data, 'package.json'));
+      t.deepEqual(d1Pjson.name, 'd');
+      t.deepEqual(d1Pjson.version, '1.0.0');
+      test.cleanup();
+    });
+test(
```
Comment (kjin) [inline test/test-package-tree.ts]: Can you add a space between separate tests?

File: test/test-package-tree.ts [hunk 3]
```diff
@@ -70,23 +66,168 @@ test(
...
+      const b1 = {name: 'b', version: '1.0.0', data: null, dependencies: [c1]};
+      const a1 = {name: 'a', version: '1.0.0', data: null, dependencies: [b1]};
+      const root =
+          {name: 'root', version: '1.0.0', data: null, dependencies: [a1]};
+      const resolvedTree: tree.PackageTree<string> =
+          await tree.resolvePaths(root, testPath);
+      const rootDependencies = resolvedTree.dependencies.sort(sortByName);
+      const a1Pjson =
+          require(path.join(rootDependencies[0].data, 'package.json'));
+      t.deepEqual(a1Pjson.name, 'a');
```
Comment (kjin) [inline test/test-package-tree.ts]: Would it be possible to do some of these checks programmatically? It seems like the three tests have quite a bit of redundant code.

File: test/test-package-tree.ts (hunk 1 above)
Comment (lee-tammy) [reply test/test-package-tree.ts]: done

File: test/test-package-tree.ts (hunk 2 above)
Comment (lee-tammy) [reply test/test-package-tree.ts]: done

File: test/test-package-tree.ts (hunk 3 above)
Comment (lee-tammy) [reply test/test-package-tree.ts]: I made some helper functions to prevent redundancy

File: test/test-package-tree.ts [hunk 4]
```diff
@@ -407,10 +348,27 @@ test(
...
+function createPackageMap(
+    pkg: tree.PackageTree<string>, map: Map<string, tree.PackageTree<string>>) {
+  pkg.dependencies.forEach((p) => {
+    createPackageMap(p, map);
+  });
+  map.set(`${pkg.name}@${pkg.version}`, pkg);
+  return map;
+}
+
+function checkPjsons<T>(
```
Comment (kjin) [inline test/test-package-tree.ts]: Nit: Can you add a description of what this function is used for?

**Review by kjin — APPROVED** [review]
LGTM w/ nit.

File: test/test-package-tree.ts (hunk 4 above)
Comment (lee-tammy) [reply test/test-package-tree.ts]: done

**Review by kjin — APPROVED** [review]

**Pull request merged**